File: circle.py
import math
import time

PI = math.acos(-1)

def loop_shape(total_steps, radius, offset):
    i = 0
    while i <= total_steps:
        theta = (i / total_steps) * 2.0 * PI + offset
        x = radius * math.cos(theta)
        y = radius * math.sin(theta)
        i += 1

with open("python_results.csv", "w") as outfile:
    outfile.write("steps,time_ms\n")
    steps = 100
    while steps <= 1000000:
        start = time.perf_counter()
        loop_shape(steps, 10, 0.0)
        end = time.perf_counter()
        ms = (end - start) * 1000
        outfile.write(f"{steps},{ms:.6f}\n")
        steps += 1000

File: circle.cpp
#include <iostream>
#include <fstream>
#include <iomanip>
#include <cmath> 
#include <chrono>

const double PI = std::acos(-1);
void loop_shape(int total_steps, int radius, double offset) {

    int i = 0;
    while (i <= total_steps){

        
        double theta = (i / (double)total_steps) * 2.0 * PI + offset;

        double x = radius * std::cos(theta);
        double y = radius * std::sin(theta);

    i += 1;
    }
}

int main(){
    std::ofstream outfile("results.csv");
    outfile << std::fixed << std::setprecision(6);
    outfile << "steps,time_ms\n";

    for (int steps = 100; steps <= 1000000; steps += 1000) {
        auto start = std::chrono::high_resolution_clock::now();
        loop_shape(steps, 10, 0.0);
        auto end = std::chrono::high_resolution_clock::now();

        double ms = std::chrono::duration<double, std::milli>(end - start).count();
        outfile << steps << "," << ms << "\n";
    }
    return 0;
}